central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -228,9 +348,32 @@ func (ds *datastoreImpl) DryRunAddCollection(ctx context.Context, collection *st
 		return err
 	}
 
+	if collection == nil {
+		return errors.New("passed collection must not be nil")
+	}
+
 	return ds.dryRunAddCollectionInGraph(collection)
 }
 
+func (ds *datastoreImpl) UpdateCollection(ctx context.Context, collection *storage.ResourceCollection) error {
+	if collection == nil {
+		return errors.New("passed collection must not be nil")
+	}
+
+	// update graph first to detect cycles
+	edgeUpdates, err := ds.updateCollectionInGraph(collection)
+	if err != nil {
+		return err
+	}
```

Comment:
not for now but this is essentially a variation to update dryrun. Dryrun is test changes + revert changes instead of commit.